### PR TITLE
gruvbox-gtk-theme: init at unstable-2022-12-09

### PR DESCRIPTION
**Pull request by Math-42**
###### Description of changes

Add a gruvbox theme for the gnome shell: 
Home page: https://github.com/Fausto-Korpsvart/Gruvbox-GTK-Theme
See also: https://www.pling.com/p/1681313/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, gtk-engine-murrine
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:4]:
```suggestion
, gnome-themes-extra
, gtk-engine-murrine
```

It is mentioned in the [README.md](https://github.com/Fausto-Korpsvart/Gruvbox-GTK-Theme#installing-themes) file:

> Before applying the themes you need to check if the Murrine Engine and gnome-themes-extra or gnome-themes-standard packages are already installed on your distro, as it is necessary for the correct rendering of the themes.

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}:
+stdenv.mkDerivation rec {
+  pname = "gruvbox-gtk-theme";
+  version = "unstable-2022-12-09";
+  name = "${pname}-${version}";
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:9]:
```suggestion
```
No need to have this, as it is the default.

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Fausto-Korpsvart";
+    repo = "Gruvbox-Gtk-Theme";
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:13]:
```suggestion
    repo = pname;
```

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    rev = "c3172d8dcba66f4125a014d280d41e23f0b95cad";
+    sha256 = "1411mjlcj1d6kw3d3h1w9zsr0a08bzl5nddkkbv7w7lf67jy9b22";
+  };
+
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:17]:
```suggestion

  buildInputs = [
    gnome-themes-extra
  ];

```

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, stdenv
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:2]:
```suggestion
, stdenvNoCC
```
No need to use a stdenv with a full blown compilation system.

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+, fetchFromGitHub
+, gtk-engine-murrine
+}:
+stdenv.mkDerivation rec {
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:6]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

**Review by romildo — CHANGES_REQUESTED** [review]

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+, gnome-themes-extra
+, gtk-engine-murrine
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "gruvbox-gtk-theme";
+  version = "unstable-2022-12-09";
+
+  src = fetchFromGitHub {
+    owner = "Fausto-Korpsvart";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix:13]: Use the exact same name here used in the github project url.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix (hunk 7 above)
Comment (Math-42) [reply pkgs/data/themes/gruvbox-gtk-theme/default.nix:13]: Ah yes, I thought that was the case, but I was recommended to switch to pname.

File: pkgs/data/themes/gruvbox-gtk-theme/default.nix [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+, gnome-themes-extra
+, gtk-engine-murrine
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "gruvbox-gtk-theme";
+  version = "unstable-2022-12-09";
+
+  src = fetchFromGitHub {
+    owner = "Fausto-Korpsvart";
+    repo = " Gruvbox-GTK-Theme";
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-gtk-theme/default.nix]:
```suggestion
    repo = "Gruvbox-GTK-Theme";
```

**Review by romildo — CHANGES_REQUESTED** [review]

**Pull request merged**